samcli/commands/check/lib/command_context.py

```diff
@@ -0,0 +1,89 @@
+import os
+import functools
+
+import click
+import boto3
+from boto3.session import Session
+
+from samcli.commands.check.command import LOG
+from samtranslator.translator.managed_policy_translator import ManagedPolicyLoader
+from samtranslator.translator.translator import Translator
+from samtranslator.public.exceptions import InvalidDocumentException
+from samtranslator.parser import parser
+
+from samcli.commands.local.cli_common.user_exceptions import SamTemplateNotFoundException
+from samcli.yamlhelper import yaml_parse
+
+from samcli.lib.replace_uri.replace_uri import _replace_local_codeuri as external_replace_local_codeuri
+from ..exceptions import InvalidSamDocumentException
+
+
+class CheckContext:
+    """
+    This class translates a template (SAM or CFN json) into a CFN yaml format. Evenchually
+    this class will also contain the major function calls for sam check, such as
+    "ask_bottle_neck_quesions", "calculate_bottle_necks", "calculate_pricing", and
+    "print_results"
+    """
+
+    def __init__(self, region, profile, template_path):
+        self.region = region
+        self.profile = profile
+        self.template_path = template_path
+
+    def run(self):
+        converted_template = self.transform_template()
+
+        click.echo("... analyzing application template")
```

Comment:
Let's don't use click as much as we can in `lib` folder. I would prefer to use `LOG.info` for now.